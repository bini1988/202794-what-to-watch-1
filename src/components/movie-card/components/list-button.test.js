import React from "react";
import renderer from "react-test-renderer";
import ListButton from "./list-button";
import card from "../../../mocks/movie-card";

const Nop = () => {};

it(`ListButton correctly renders default markup`, () => {
  const tree = renderer.create(
      <ListButton
        card={card}
        onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`ListButton correctly renders in list markup`, () => {
  const tree = renderer.create(
      <ListButton
        card={{...card, isInList: true}}
        onClick={Nop}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
