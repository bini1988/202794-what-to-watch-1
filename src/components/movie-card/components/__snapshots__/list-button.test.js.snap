// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListButton correctly renders default markup 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height={20}
    viewBox="0 0 19 20"
    width={19}
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`ListButton correctly renders in list markup 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height={14}
    viewBox="0 0 18 14"
    width={18}
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
